assets/src/domain/types.ts

```diff
@@ -0,0 +1,114 @@
+import type { EventEditorData } from '@edtrInterfaces/types';
+import type { WpPluginsPageData } from './wpPluginsPage/types';
+import type {
+	CurrencyProps,
+	CurrentUserProps,
+	GeneralSettings,
+	LocaleProps,
+	SiteUrlProps,
+} from '@application/valueObjects/config/types';
+
+/**
+ * Contains the API related information
+ */
+export interface ApiData {
+	nonce?: string;
+	restBaseUrl?: string;
+	graphqlEndpoint?: string;
+}
+
+/**
+ * This can contain asset information like
+ * - base URL to assets directory
+ * - URLs to specific images like brand images etc.
+ */
+export interface AssetsData {
+	url: string; // URL to assets directory
+}
+
+export interface ConfigData {
+	// Better to name the type as "Currency" instead of "CurrencyProps"
+	currency?: CurrencyProps;
+	// Better to name the type as "CurrentUser" or just "User" instead of "CurrentUserProps"
+	currentUser?: CurrentUserProps;
+	generalSettings?: GeneralSettings;
+	locale?: LocaleProps;
+	siteUrl?: SiteUrlProps;
+}
+
+export interface I18nInfo {
+	domain: string; // e.g. "event_espresso"
+	lang: string; // e.g. "en_US"
+	plural_forms?: string;
+}
+
+export type I18nData = {
+	'': I18nInfo;
```

Comment:
what is this?

Reply:
That's how wp i18n expects the data, the empty string key contains the locale information.